services/catalog/src/simcore_service_catalog/schemas/schemas_dags.py

```diff
@@ -0,0 +1,42 @@
+from typing import Dict, Optional
+
+# TODO: why pylint error in pydantic???
+# pylint: disable=no-name-in-module
+from pydantic import BaseModel, EmailStr, Field, Json
+
+from . import project
+from .project import KEY_RE, VERSION_RE
+
+
+class DAGBase(BaseModel):
+    key: str = Field(
+        ..., regex=KEY_RE, example="simcore/services/frontend/nodes-group/macros/1"
+    )
+    version: str = Field(..., regex=VERSION_RE, example="1.0.0")
+    name: str
+    description: Optional[str]
+    contact: Optional[EmailStr]
+
+
+class DAGIn(DAGBase):
+    workbench: Optional[Dict[str, project.Node]]
+
+
+class DAGInPath(DAGBase):
```

Comment:
will need some info from you as to what this is...

Reply:
This is something amazingly cool :-) we can talk about it on Monday